### ISSUE TITLE
Generic and subscriptable annotations result in errors

### ISSUE THREAD

**Issue by HeapUnderfl0w** [RivenSkaye/EnforceTypes#3]
Using a function with a `Any` type parameter does not work and crashes instead

## MVE

```python
from typing import Any
import EnforceTypes as ET

@ET.functypes
def mve(params: Any):
    pass # function content is irrelevant

mve("funny param")
```

this results in the following error:

```
Traceback (most recent call last):
  File "E:\files\source\Projects\_misc\ex\any_example_enforcetypes\mve.py", line 8, in <module>
    mve("funny param")
  File "C:\Users\HeapUnderflow\.virtualenvs\any_example_enforcetypes-0Csrbawq\lib\site-packages\EnforceTypes\Functions.py", line 42, in newcall
    if not isinstance(kw_val, argtype) and not isinstance(kw_val, argtuple):
  File "C:\Program Files\Python39\lib\typing.py", line 345, in __instancecheck__
    raise TypeError(f"{self} cannot be used with isinstance()")
TypeError: typing.Any cannot be used with isinstance()
```

interesting enough using a `list->tuple->any` chain exhibits slightly different behavior

```python
from typing import Any, List, Tuple
import EnforceTypes as ET

@ET.functypes
def mve(params: List[Tuple[Any, Any]]):
    pass # function content is irrelevant

mve([("funny param a", 0x69420)])
```

this results in the following error:

```
Traceback (most recent call last):
  File "E:\files\source\Projects\_misc\ex\any_example_enforcetypes\mve.py", line 8, in <module>
    mve([("funny param a", 0x69420)])
  File "C:\Users\HeapUnderflow\.virtualenvs\any_example_enforcetypes-0Csrbawq\lib\site-packages\EnforceTypes\Functions.py", line 42, in newcall
    if not isinstance(kw_val, argtype) and not isinstance(kw_val, argtuple):
  File "C:\Program Files\Python39\lib\typing.py", line 715, in __instancecheck__
    return self.__subclasscheck__(type(obj))
  File "C:\Program Files\Python39\lib\typing.py", line 718, in __subclasscheck__
    raise TypeError("Subscripted generics cannot be used with"
TypeError: Subscripted generics cannot be used with class and instance checks
```

# System Information

OS: Windows 10, x64
Python: 3.9.7 & 3.9.13

**Comment by HeapUnderfl0w** [comment]
Pipenv config used to configure the environment for the MVE

```toml
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
enforcetypes = "*"

[dev-packages]

[requires]
python_version = "3.9"
```

and then all mve's where run in a `pipenv shell` using `python ./mve.py`

**Comment by RivenSkaye** [comment]
Forgetting to check for `Any` is an oversight on my end, I should just skip any arg hinted that way.
The issue with `Generics` and `subscripted` classes is a fun issue though. I'll work on that

**Comment by RivenSkaye** [comment]
Could you see if [this commit](https://github.com/RivenSkaye/EnforceTypes/commit/1c0ffee8f9cba3654161af05f4c1adb639f5dbe3) fixes your issue? That's the latest on the v0.1.0 branch
It looks like it, but it also makes it clear that I currently don't offer active support for checking types within collections, e.g. verifying a `List[int]` actually only contains ints

**Comment by RivenSkaye** [comment]
Make that one coffee later, I forgot to push a dependency upgrade 🤦🏻‍♂️
